src/lib/rest/StringFilter.cpp

```diff
@@ -1323,7 +1698,7 @@ bool StringFilter::match(ContextElementResponse* cerP)
 */
 StringFilter* StringFilter::clone(std::string* errorStringP)
 {
-  StringFilter* sfP = new StringFilter();
+  StringFilter* sfP = new StringFilter(type);
```

Comment:
Discussed over skype.
However, I can mention it here also ...

As there are now two flavors of StringFilter ('q' and 'mq'), when creating an instance of the class, the instance must be initialized to indicate the flavor. 

NTC